api/chat.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"log"
+
+	"github.com/gin-gonic/gin"
+	"github.com/goccy/go-json"
+	"github.com/mhghw/fara-message/db"
+)
+
+func NewChatAPI(c *gin.Context) {
+	var requestBody NewChat
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+		log.Print("failed to bind json", err)
+		return
+	}
+	if err := db.NewChat(requestBody.ChatName, requestBody.ChatType, requestBody.Users); err != nil {
```

Comment:
you also need to separate endpoints for chat types, client shouldn't specify the chat type for you, also direct messages has no chatName